src/datatypes/src/vectors.rs

```diff
@@ -174,8 +174,19 @@ pub trait MutableVector: Send + Sync {
 
     /// Push value ref to this mutable vector.
     ///
-    /// Returns error if data types mismatch.
-    fn push_value_ref(&mut self, value: ValueRef) -> Result<()>;
+    /// # Panics
+    /// Panics if error if data types mismatch.
+    fn try_push_value_ref(&mut self, value: ValueRef) -> Result<()>;
+
+    /// Push value ref to this mutable vector.
+    fn push_value_ref(&mut self, value: ValueRef) {
+        self.try_push_value_ref(value).unwrap()
```

Comment:
```suggestion
        self.try_push_value_ref(value).unwrap_or_else(|| format!("expecting pushing value of datatype {}, actual {}", self.data_type(), value.data_type()))
```

Reply:
Great, thank you for the suggestion, Michael.
Seems that a didn't fully understand this bit from requirement  "changing the `push_value_ref()` to panic by default".
I'll push the change later.

Reply:
@MichaelScofield
This is the version of `push_value_ref` I came up with

```rust
   fn push_value_ref(&mut self, value: ValueRef) {
        self.try_push_value_ref(value).unwrap_or_else(|_| {
            format!(
                "expecting pushing value of datatype {:?}, actual {:?}",
                self.data_type(),
                value
            );
        });
    }
```
Two (probably noob) questions:
1) returned sting from `format!` is ignored as the current implementation returns unit.
Should I switch to `panic!` instead of ` format!` maybe?
2) in the `format!` itself will it be fine to use `value` as is?
It seems to give a meaningful result  
`expecting pushing value of datatype Binary(BinaryType), actual Int32(123)`

Reply:
>Should I switch to panic! instead of  format! maybe?

+1. Otherwise this method won't actually panic and may result in data corruption.

>in the format! itself will it be fine to use value as is?

Yes, `ValueRef` is an enum and its inner value will be printed out, which is beneficial for debugging.